pkg/patch/patch.go

```diff
@@ -221,18 +241,22 @@ func patchSingleArchImage(
 	}
 
 	// if the target platform is different from the host platform, we need to check if emulation is enabled
-	osEqual := platforms.DefaultSpec().OS == targetPlatform.OS
-	archEqual := platforms.DefaultSpec().Architecture == targetPlatform.Architecture
-	if osEqual && archEqual {
-		log.Debugf("Host platform %+v matches target platform %+v", platforms.DefaultSpec(), targetPlatform)
-	} else {
-		log.Debugf("Host platform %+v does not match target platform %+v", platforms.DefaultSpec(), targetPlatform)
-		// check if emulation is enabled
+	// only need to do this check if were patching a multi-arch image
+	if multiArch {
+		hostPlatform := platforms.DefaultSpec()
+		platformsEqual := hostPlatform.OS == targetPlatform.OS &&
```

Comment:
in single arch we set linux os, should we do the same here?

Reply:
thanks for catching that, yes it should, I've added it in